### PR TITLE
[complex32] make_tensor

### PR DESCRIPTION
**Pull request by kshitij12345**
Update `make_tensor` so that it can generate `complex32` tensor.

**Note**: This doesn't enable `complex32` tests in the OpInfo test suite but only updates `make_tensor` to generate it. Enabling `complex32` in the test suite will be done later PRs

### CODE REVIEW

File: torch/testing/_creation.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 import math
 import collections.abc
 
+complex_float_map = {torch.complex32: torch.half, torch.complex64: torch.float, torch.complex128: torch.double}
```
Comment (mruberry) [inline torch/testing/_creation.py:10]: Prefix private datastructures with an underscore and add a comment explaining what this map is for

File: torch/testing/_creation.py [hunk 2]
```diff
@@ -126,7 +127,7 @@ def clamp(a, l, h):
         rand_val = torch.rand(shape, device=device, dtype=dtype)
         result = high * rand_val + low * (1 - rand_val)
     elif dtype in _complex_types:
-        float_dtype = torch.float if dtype is torch.cfloat else torch.double
+        float_dtype = complex_float_map[dtype]
```
Comment (mruberry) [inline torch/testing/_creation.py:130]: Name suggestion: `_complex_to_corresponding_float_type_map`

**Review by mruberry — COMMENTED** [review]
Two small inline comments -- just ping me when they're addressed and we'll get this merged

**Review by pmeier — APPROVED** [review]
Minor comments inline. Otherwise LGTM, thanks @kshitij12345!

File: torch/testing/_creation.py (hunk 2 above)
Comment (kshitij12345) [reply torch/testing/_creation.py]: Looks good. Done. Thanks!

File: torch/testing/_creation.py [hunk 3]
```diff
@@ -106,7 +106,7 @@ def clamp(a, l, h):
 
     _integral_types = [torch.uint8, torch.int8, torch.int16, torch.int32, torch.int64]
     _floating_types = [torch.float16, torch.bfloat16, torch.float32, torch.float64]
-    _complex_types = [torch.cfloat, torch.cdouble]
```
Comment (anjali411) [inline torch/testing/_creation.py:109]: might be nice to add an alias called `chalf` to complex32 in the future

File: torch/testing/_creation.py [hunk 4]
```diff
@@ -126,7 +126,9 @@ def clamp(a, l, h):
         rand_val = torch.rand(shape, device=device, dtype=dtype)
         result = high * rand_val + low * (1 - rand_val)
     elif dtype in _complex_types:
-        float_dtype = torch.float if dtype is torch.cfloat else torch.double
+        float_dtype = {torch.complex32: torch.float16,
```
Comment (anjali411) [inline torch/testing/_creation.py:129]: IIRC, this same logic was also being used in an earlier PR. might make sense to add a helper function?

**Review by anjali411 — COMMENTED** [review]
might be nice to have a helper fn or a global map but looks good otherwise.

File: torch/testing/_creation.py [hunk 5]
```diff
@@ -7,6 +7,11 @@
 import math
 import collections.abc
 
+# Used by make_tensor for generating complex tensor.
+complex_to_corresponding_float_type_map = {torch.complex32: torch.float16,
```
Comment (kshitij12345) [inline torch/testing/_creation.py:11]: Was planning to put this in `torch.testing._internal.common_utils.py` but it uses `make_tensor`. Wasn't ready to fight circular dependency. So have kept it here.

**Review by anjali411 — APPROVED** [review]
@pytorchbot merge this please

**Pull request closed without merging**